twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillController.java

```diff
@@ -77,7 +77,7 @@
    */
   YarnTwillController(String appName, RunId runId, ZKClient zkClient,
                       final ApplicationMasterLiveNodeData amLiveNodeData, final YarnAppClient yarnAppClient) {
-    super(runId, zkClient, Collections.<LogHandler>emptyList());
+    super(appName, runId, zkClient, amLiveNodeData.getKafkaZKConnect() != null, Collections.<LogHandler>emptyList());
```

Comment:
At this point is more about style than correctness. I personally prefer to not make the logic of determining to enable or disable logging in YarnTwillController but rather on object that have current state of the union.